cts-java/src/test/java/org/ga4gh/cts/api/references/ReferencesSearchIT.java

```diff
@@ -22,41 +23,97 @@
-     * ncbiTaxonId == 9606 (human)</pre></li>
-     * <li>Query 2: <pre>/references/(ref ID)</pre></li>
-     * <li>Test 2: assert that the returned {@link Reference} has <pre>ID == ref ID</pre></li>
-     * </ul>
+     * Fetch all references in the reference set, check there are some.
+     *
      * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
      */
     @Test
-    public void searchForExpectedReferences() throws AvroRemoteException {
+    public void checkSearchingAllReferences() throws AvroRemoteException {
+        final String referenceSetId = 
+            Utils.getReferenceSetIdByAssemblyId(client, TestData.REFERENCESET_ASSEMBLY_ID);
+        final SearchReferencesRequest req =
+                SearchReferencesRequest.newBuilder()
+                        .setReferenceSetId(referenceSetId)
+                        .build();
+        final SearchReferencesResponse resp = client.references.searchReferences(req);
+
+        final List<Reference> refs = resp.getReferences();
+        assertThat(refs).isNotEmpty();
+    }
 
-//        final String expectedMd5 = "1b22b98cdeb4a9304cb5d48026a85128";
-        final String expectedMd5 = "TODO";
-        final int expectedLength = 8050;
-//        final int expectedLength = 249250621;
-        final int expectedTaxonId = 9606;
+    /**
+     * Fetch reference by accession.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkSearchReferencesByAccessionNumber() throws AvroRemoteException {
+        final String referenceSetId =
+                Utils.getReferenceSetIdByAssemblyId(client, TestData.REFERENCESET_ASSEMBLY_ID);
+        final String accession = "GA4GH_CTS_01_BRCA1";
+        final String referenceName = "ref_brca1";
+        final SearchReferencesRequest req =
+                SearchReferencesRequest.newBuilder()
+                        .setReferenceSetId(referenceSetId)
+                        .setAccessions(aSingle(accession))
+                        .build();
+        final SearchReferencesResponse resp = client.references.searchReferences(req);
+
+        final List<Reference> refs = resp.getReferences();
+        assertThat(refs).isNotEmpty();
+        refs.stream().forEach(rs -> StrictAssertions.assertThat(rs.getName()).isEqualTo(referenceName));
+    }
+
+    /**
+     * Check that no reference is fetched on an invalid accession.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkSearchReferencesInvalidAccession() throws AvroRemoteException {
+        final String referenceSetId =
+                Utils.getReferenceSetIdByAssemblyId(client, TestData.REFERENCESET_ASSEMBLY_ID);
+        final String accession = "QUUX";
```

Comment:
We can't 100% guarantee that it's invalid, so I didn't want the name to imply that we can.
We can guarantee that it's randomly generated, though.